docs/rfcs/001-updateable-bundled-packages.md

```diff
@@ -0,0 +1,100 @@
+
+`apm` will be updated to include the list of bundled packages with `"updateable": true` set in their `package.json` so that the user will be notified of new package versions that support the engine version of their current Atom build.
+
+### User Experience Examples
+
+1. The user downloads Atom 1.28.0 from atom.io which includes GitHub package version 0.15.0.  After Atom 1.28.0 was released, a hotfix release was shipped for the GitHub package as 0.15.1.  When the user installs and starts Atom, they are prompted to install the update to the GitHub package.
+
+2. The user downloads and installs Atom 1.28.0 from atom.io which includes GitHub package version 0.15.0.  Two weeks later, GitHub package 0.16.0 is released with a few new features.  The user is prompted to update to the new version and gets the new features even though Atom 1.29.0 hasn't been released yet.
+
+3. In the future, a user has an old install of Atom 1.28.0 and waits a long time between installing Atom updates.  The GitHub package releases version 0.25.0 but the user is not prompted to install it because the GitHub package has set `engines` in `package.json` to restrict to Atom 1.32.0 and above.
+
+### Rules for Updateable Bundled Packages
+
+Any package that opts into this behavior must follow one rule: **its `engines` field must be regularly updated to reflect the necessary Atom version for the Atom, Electron, and Node.js APIs used in the package**.  This field defines the range of Atom versions in which the package is expected to work.  The field should always be set to the lowest possible Atom version that the package supports.
+
+If a package wants to use API features of a newer version of Atom while still supporting older Atom versions, it must do so in a way that is aware of the user's version and adjust itself accordingly.
+
+## Drawbacks
+
+> Why should we *not* do this?
+
+The primary drawback of this approach is that updateable bundled packages might exhibit problems on older Atom versions due to missing or changed APIs in Atom, Electron, or Node.js.  The solution for these packages is to keep their `engines` field updated appropriately, but there's still a chance that some updates will slip through without the necessary engine version changes.
+
+One other possible drawback is that an updated version of a bundled package might not be compatible across two different Atom channels.  For example, if the user installs a new update to a bundled package that only supports the current Atom Beta release or higher, the user will no longer have access to that package if they open Atom Stable.
+
+However, this drawback is no different than what the user would face today installing a community package under the same circumstances, so this could be considered a general problem in the Atom package ecosystem.
```

Comment:
This sentence seems to apply to the preceding paragraph. Would it make sense to move this sentence to end of the preceding paragraph?

Reply:
Yep, originally it was part of that paragraph but it felt like the paragraph was too long.  I'll join them back together.  :heavy_check_mark: